base/math.jl

```diff
@@ -832,7 +832,7 @@ _hypot(x::ComplexF16, y::ComplexF16) = Float16(_hypot(ComplexF32(x), ComplexF32(
 
 function _hypot(x::NTuple{N,<:Number}) where {N}
     maxabs = maximum(abs, x)
-    if isnan(maxabs) && any(isinf, x)
+    if any(isinf, x)
```

Comment:
Why this particular change? For cases like `_hypot((NaN, Inf))`, the result would currently be `NaN`, but with this change, it would be `Inf`. I guess the argument for changing it is that if we interpret `NaN` as stand-in for some unknown number, then if any of `x` is `Inf`, there is no value that we could substitute for `NaN` that would make the result not `Inf`. But `NaN` is defined as "an indeterminate value which is neither an infinity nor a finite number", so this doesn't seem like a valid interpretation.

Reply:
IEEE754 defines `hypot(Inf,NaN) == Inf`. Try interpreting `NaN` as "an unknown real number or infinity" and you'll see that the semantics make sense.

This change doesn't do what you think it does. Take a careful look and you'll see that the `isnan` check only served to skip the `any(isinf,...)` check when `maxabs` isn't `NaN`, because the following line will catch it anyway. But what I have messed up here is that with this change, the `isinf(maxabs)` check that immediately follows is redundant.

But on second thought, I'll revert this change anyway. I don't think that particular method will get called very often since the new `IEEEFloat` version should very often shadow it. Perhaps with whatever exotic types do make it to this method (note that `BigFloat` also has its own method) the short-circuiting may be helpful.